testplan/report/testing/base.py

```diff
@@ -297,17 +303,20 @@ def bubble_up_attachments(self):
         """
         for child in self:
             if getattr(child, 'fix_spec_path', None):
-                real_path = child.fix_spec_path
-                hash_dir = hashlib.md5(real_path.encode('utf-8')).hexdigest()
-                # I think the basename should be slugified?
-                hash_path = os.path.join(
-                    hash_dir,
-                    os.path.basename(child.fix_spec_path)
-                )
-                child.fix_spec_path = hash_path
-                self.attachments[hash_path] = real_path
-
-            # Add logic to find result.attach assertions.
+                self._bubble_up_fix_spec(child)
+            for attachment in child.attachments:
+                self.attachments[attachment.dst_path] = attachment.source_path
+
+    def _bubble_up_fix_spec(self, child):
+        """Bubble up a "fix_spec_path" from a child report."""
+        real_path = child.fix_spec_path
+        hash_dir = hashlib.md5(real_path.encode('utf-8')).hexdigest()
```

Comment:
Can you clarify what this code is doing?
What I understand at the moment is that the child.fix_spec_path is location of the TestGroupReport?
And the hash_path is where the UI will locate the attachment?

Reply:
To be honest I'm not 100% sure, I mostly just took the existing code and moved it into a sub-function. I _think_ that the fix_spec_path is a path to a file that is specific to the FIX protocol. FIX is a protocol used in trading, it can have multiple encodings and custom tag values so I think the spec file has something to do with that.